test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -1386,6 +1386,31 @@ public void HubsCannotHaveOverloadedMethods()
             }
         }
 
+        [Fact]
+        public async Task ReceiveCorrectErrorFromStreamThrowing()
+        {
+            var serviceProvider = HubEndPointTestUtils.CreateServiceProvider();
+            var endPoint = serviceProvider.GetService<HubEndPoint<StreamingHub>>();
+
+            using (var client = new TestClient())
+            {
+                var endPointLifetime = endPoint.OnConnectedAsync(client.Connection);
+
+                await client.Connected.OrTimeout();
+
+                var messages = await client.StreamAsync(nameof(StreamingHub.ThrowStream));
+
+                Assert.Equal(1, messages.Count);
+                var completion = messages[0] as CompletionMessage;
+                Assert.NotNull(completion);
+                Assert.Equal("Exception from observable", completion.Error);
+
+                client.Dispose();
```

Comment:
Do you need to dispose here? You have the using statement

Reply:
Needed because `endPointLifetime` is blocked until the client goes away